From 3e49a0e0d3557aa1737e2bbd974883245b82b9ae Mon Sep 17 00:00:00 2001
From: Konstantin Pavlov <thresh@nginx.com>
Date: Wed, 22 Nov 2023 13:34:29 -0800
Subject: [PATCH] Partially revert "THRIFT-4861: Set minimum boost to 1.56 and
 stop using deprecated boost predef endian header"

This reverts parts of commit ee8caca54ef51e05c7eee8d67e3c63cd73081e8a.
---
 build/cmake/BoostMacros.cmake           |  2 +-
 lib/cpp/src/thrift/protocol/TProtocol.h | 17 +++++++----------
 2 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/build/cmake/BoostMacros.cmake b/build/cmake/BoostMacros.cmake
index ffb85af32..c99073885 100644
--- a/build/cmake/BoostMacros.cmake
+++ b/build/cmake/BoostMacros.cmake
@@ -17,7 +17,7 @@
 # under the License.
 #
 
-set(BOOST_MINREV 1.56)
+set(BOOST_MINREV 1.53)
 
 macro(REQUIRE_BOOST_HEADERS)
   find_package(Boost ${BOOST_MINREV} QUIET REQUIRED)
diff --git a/lib/cpp/src/thrift/protocol/TProtocol.h b/lib/cpp/src/thrift/protocol/TProtocol.h
index c6ff36049..e1241c86b 100644
--- a/lib/cpp/src/thrift/protocol/TProtocol.h
+++ b/lib/cpp/src/thrift/protocol/TProtocol.h
@@ -106,18 +106,15 @@ static inline To bitwise_cast(From from) {
 #  define __THRIFT_LITTLE_ENDIAN LITTLE_ENDIAN
 #  define __THRIFT_BIG_ENDIAN BIG_ENDIAN
 # else
-#  include <boost/predef/other/endian.h>
-#  if BOOST_ENDIAN_BIG_BYTE
-#    define __THRIFT_BYTE_ORDER 4321
-#    define __THRIFT_LITTLE_ENDIAN 0
-#    define __THRIFT_BIG_ENDIAN __THRIFT_BYTE_ORDER
-#  elif BOOST_ENDIAN_LITTLE_BYTE
-#    define __THRIFT_BYTE_ORDER 1234
-#    define __THRIFT_LITTLE_ENDIAN __THRIFT_BYTE_ORDER
-#    define __THRIFT_BIG_ENDIAN 0
-#  endif
+#  include <boost/config.hpp>
+#  include <boost/detail/endian.hpp>
+#  define __THRIFT_BYTE_ORDER BOOST_BYTE_ORDER
 #  ifdef BOOST_LITTLE_ENDIAN
+#   define __THRIFT_LITTLE_ENDIAN __THRIFT_BYTE_ORDER
+#   define __THRIFT_BIG_ENDIAN 0
 #  else
+#   define __THRIFT_LITTLE_ENDIAN 0
+#   define __THRIFT_BIG_ENDIAN __THRIFT_BYTE_ORDER
 #  endif
 # endif
 #endif
-- 
2.39.3 (Apple Git-145)

